lib/rules/rule-nested-empty-line-before/__tests__/index.js

```diff
@@ -122,6 +122,40 @@ testRule(rule, {
 
 testRule(rule, {
   ruleName,
+  config: [ "always", { except: ["after-rule"] } ],
```

Comment:
These tests will likely fail when you update the logic.

Can you also include a test (both passing and failing) for when a rule is preceded by an at-rule e.g.

```
a {
  @media {
    b {}
  }
  
  d {}
  d {}
}
```